### DISCUSSION TITLE
Notes on ergonomics

### DISCUSSION THREAD

**Discussion by sulivanShu** [Nain57/Smart-AutoClicker#61] [General]
Hello, this is my first time using an autoclicker.

Which would help new users:

1. on first launch, a welcome message and a link to tutorials (for example on the github wiki)
2. At first, I thought the events were triggered in chronological order. In fact, they are triggered as soon as the conditions are right. This implies that a previous event can be triggered later, and several times. This is undoubtedly an expected and desirable feature, because it avoids copying events several times. However, this should be specified somewhere, because it is then necessary to specify the conditions with more attention.
3. On applications like Google ads or Microsoft ads, it happens that only one part of the page loads, and other parts of the page load later. Events can then be triggered unexpectedly. A suitable solution is to add a delay at the end of the preceding event, in order to ensure that the page is fully loaded before checking if the conditions are met and triggering the corresponding event.
4. [The details on scenario detection are useful](https://github.com/Nain57/Smart-AutoClicker/discussions/46). it helps to understand how the program works.
5. When configuring an event, the "executions before stopping the scenario" field is invisible without scrolling, and the scroll bar is invisible. I discovered this field after several hours. It might be useful to modify the presentation to make sure that all fields are visible, or that they can be easily discovered, on different screen sizes.
6. I assume that the scenarios are saved in the application as text files. It could be useful to have a scenario import and export function, to save them, move them to other phones or edit them with a text editor. It could also be useful to be able to copy and paste a scenario, when you want to produce a variation of a previous scenario.
7. I will be using your app a lot to activate or deactivate my Google ads and Microsoft ads campaigns! It might be useful to be able to place desktop shortcuts to launch scenarios directly from the desktop.

Thank you very much for your application! This saves me from going through the Google ads and Microsoft ads APIs which I find very difficult to use! Especially to do something as simple as activating or deactivating all my campaigns at the same time!

**Comment by Nain57** [comment]
Hi ! Thank you for your feedback, there is a lot of good ideas here.

* 1 & 2 & 4: I definitely want to add some tutorial/documentation within the app to help the user. It is on my todo list, but I want to finish some core features first.
* 3: Timers can be a really useful for those cases, but I don't want to automatically add them, as use cases are different for each user of Smart AutoClicker. 
* 5 & 7: Ux is definitely not my strength, and this kind of feedback is really appreciated !
* 6: Scenarios are saved in a database, and condition images are stored in the application private folder. Import/Export is frequently requested (and I really understand the usefulness of it), unfortunately, the current detection algorithm depends on your screen size. Moving a scenario from one phone to another might not work unless the screen is identical.
